common/ip-packet-requests/src/lib.rs

```diff
@@ -1,8 +1,17 @@
+use serde::{Deserialize, Serialize};
+use std::net::{Ipv4Addr, Ipv6Addr};
+
 pub mod codec;
 pub mod request;
 pub mod response;
 
-pub const CURRENT_VERSION: u8 = 3;
+pub const CURRENT_VERSION: u8 = 4;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct IPPair {
+    pub ipv4: Ipv4Addr,
+    pub ipv6: Ipv6Addr,
```

Comment:
maybe I'm missing something, but wouldn't this change require you to provide **both** ipv4 and ipv6 addresses for each request? shouldn't ipv6 be an option instead given it might not always be available?

Reply:
oh. I guess the tun device will give you one.
but I wonder if we might have a problem if the gateway's network stack does not support ipv6 (which might be the case for people running them on some crappy vps)

Reply:
That's a fair point, but I would say more for clients than for gateways, since we kind of expect IPv6 support from gateways, right? And for clients, from the types that currently work for the vpn (linux, macos, android), I think there might only be a small portion of legacy linux devices that don't have IPv6 support in their network stack. But even so, maybe making it an `Option` now rather then having to change the version later to support them is a safer approach.

Reply:
On second thought, I think this is not the place to modify this, because it only affects gateways, like you said, and I think those should have IPv6 support as a requirement. I will take a look at client side, in the other repo, as to not fail in case of lack of IPv6 support.